pandas/tests/indexes/test_numeric.py

```diff
@@ -831,3 +831,11 @@ def test_ufunc_coercions(self):
         tm.assertIsInstance(result, Float64Index)
         exp = Float64Index([0.5, 1., 1.5, 2., 2.5], name='x')
         tm.assert_index_equal(result, exp)
+
+    def test_ensure_copied_data(self):
+        data = np.array([1, 2, 3], dtype='int64')
+        idx = Int64Index(data, copy=True)
+        self.assert_numpy_array_equal(data, idx._data)
+        tm.assertIsNot(data, idx._data)
```

Comment:
this is a VERY specific test, can you make this much more general (e.g. use ``.create_data()``) to ensure that this is correct for numeric Index types. Furthermore, would be ideal to make this test for ALL Indexes (IOW put this in TestBase)